### ISSUE TITLE
Component not being "re-mounted?" after a wire:model event is passed.

### ISSUE THREAD

**Issue by wotta** [livewire/livewire#421]
**Describe the bug**
After updating from Livewire version 0.3.* I am having trouble with my components breaking when I fill in a "wired" input field.

**To Reproduce**

1. Clone the following repository: `git@github.com:wotta/insighteer.git`
2. Follow useage instruction in readme.
3. Log in the application.
4. Click on the following link: `Families`
5. Click on the `Create family` button.
6. Fill in a name.

**Expected behavior**
I expected this would work without throwing an error. ( I know sounds easy )
But in previous versions this did work as expected and somehow the 0.5.* breaks the component re-mounting. ( maybe that's not completely how it is named but I hope you get the gist of it ).

**Screencasts**
https://monosnap.com/file/OatxTtYO8XL4nmbWg9MKlcGkCxumDm

**Desktop (please complete the following information):**
 - Firefox Developer edition 72.0b7 (64-bits)
 - MacOS Mojave: 10.14.6 (18G95)

**Additional context**
This problem does not only apply to the `CreateFormComponent`. But Also my other livewire components.

**Comment by wotta** [comment]
Don't want to act rude but has this issue been looked at yet?
I will try to rewrite my livewire components. Hopefully I can fix it when thinking about the logic.

**Comment by calebporzio** [comment]
Try making `$model` a public property instead of a protected one.

Some context:
* v0.5.0 removed the ability to store things (and persist them) inside protected properties. (https://github.com/livewire/livewire/releases/tag/v0.5.0)
* v0.5.3 added the ability to store models as public properties to curb the need for storing public properties. (https://github.com/livewire/livewire/releases/tag/v0.5.3)

**Comment by calebporzio** [comment]
We've wrestled with some pretty big architectural considerations, but the API has settled down now, so there shouldn't be as many breaking changes going forward.

**Comment by wotta** [comment]
Heej @calebporzio, first off thanks for taking the time to reply!

I took a look at the 0.6 release notes and it gave me the hope that it might have been fixed.

So took the time to update and change some stuff.

From the looks of it at this moment it may seem to me that I made some architectural mistakes that no update can help.

I still need to take a closer look at the release notes that you shared.

However I started to migrate my components to better suitable way.

If I run into any issues at a later point in time after refactoring and rereading the release notes I will definitely share my findings.
